src/org/rascalmpl/library/lang/json/internal/JsonValueReader.java

```diff
@@ -52,20 +55,39 @@ public class JsonValueReader {
   private final IValueFactory vf;
   private ThreadLocal<SimpleDateFormat> format;
   private final IRascalMonitor monitor;
+  private ISourceLocation src;
+  private VarHandle posHandler;
+  private VarHandle lineHandler;
+  private VarHandle lineStartHandler;
   
   /**
    * @param vf     factory which will be used to construct values
    * @param store  type store to lookup constructors of abstract data-types in and the types of keyword fields
    */
-  public JsonValueReader(IValueFactory vf, TypeStore store, IRascalMonitor monitor) {
+  public JsonValueReader(IValueFactory vf, TypeStore store, IRascalMonitor monitor, ISourceLocation src) {
     this.vf = vf;
     this.store = store;
     this.monitor = monitor;
+    this.src = src;
     setCalendarFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+
+    if (src != null) {
+      try {
+        var lookup = MethodHandles.lookup();
+        var privateLookup = MethodHandles.privateLookupIn(JsonReader.class, lookup);
+        this.posHandler = privateLookup.findVarHandle(JsonReader.class, "pos", int.class);
+        this.lineHandler = privateLookup.findVarHandle(JsonReader.class, "lineNumber", int.class);
+        this.lineStartHandler = privateLookup.findVarHandle(JsonReader.class, "lineStart", int.class);
+      }
+      catch (NoSuchFieldException | SecurityException | IllegalAccessException e) {
+        // we disable the origin tracking if we can not get to the fields
```

Comment:
nice fallback. maybe print and error message for us to detect that this happened?